lib/handler/mruby/embedded.c.h

```diff
@@ -74,7 +74,7 @@
     "            while 1\n"                                                                                                        \
     "              resp = app.call(req)\n"                                                                                         \
     "              cached = self_fiber\n"                                                                                          \
-    "              (req, generator) = Fiber.yield(*resp, generator)\n"                                                             \
+    "              (req, generator) = Fiber.yield(resp, generator)\n"                                                              \
```

Comment:
Am I correct in assuming that in #1173 we switched from 3-arg return to 4-arg return (i.e. status + headers + body + generator), but forgot to set the 4th argument in some cases?

If that is true, why are we switching to a 3-element array + generator now? Is there a reason?

Reply:
> forgot to set the 4th argument in some cases?

No.  The source of the problem was array expansion operator (`*`). If we expand nil, it produces nothing (i.e. `[*nil]` -> `[]`). In other words, if we expand `resp` here, the position of generator in the array is undeterministic.

Reply:
Ah! Thank you for the clarification.